DuckDuckGo/Freemium/PIR/FreemiumDBPPromotionViewCoordinator.swift

```diff
@@ -0,0 +1,174 @@
+    }
+
+    /// Marks the user as onboarded in the Freemium DBP system.
+    func markUserAsOnboarded() {
+        freemiumDBPUserStateManager.didOnboard = true
+    }
+
+    /// Shows the Freemium PIR user interface via the presenter.
+    func showFreemiumPIR() {
+        freemiumPIRPresenter.showFreemiumPIR(
+            didOnboard: freemiumDBPUserStateManager.didOnboard,
+            windowControllerManager: WindowControllersManager.shared
+        )
+    }
+
+    /// Dismisses the home page promotion and updates the user state to reflect this.
+    func dismissHomePagePromotion() {
+        didDismissHomePagePromotion = true
+    }
+
+    /// Sets the initial visibility state of the promotion based on whether the promotion was
+    /// previously dismissed and whether the Freemium PIR feature is available.
+    func setInitialPromotionVisibilityState() {
+        isHomePagePromotionVisible = (!didDismissHomePagePromotion && freemiumPIRFeature.isAvailable)
+    }
+
+    /// Creates the view model for the promotion, updating based on the user's scan results.
+    ///
+    /// - Returns: The `PromotionViewModel` that represents the current state of the promotion.
+    func createViewModel() -> PromotionViewModel {
+        if let results = freemiumDBPUserStateManager.firstScanResults {
+            if results.matchesCount > 0 {
+                return .freemiumPIRPromotionScanEngagementResults(
+                    resultCount: results.matchesCount,
+                    brokerCount: results.brokerCount,
+                    proceedAction: proceedAction,
+                    closeAction: closeAction
+                )
+            } else {
+                return .freemiumPIRPromotionScanEngagementNoResults(
+                    proceedAction: proceedAction,
+                    closeAction: closeAction
+                )
+            }
+        } else {
+            return .freemiumPIRPromotion(proceedAction: proceedAction, closeAction: closeAction)
+        }
+    }
+
+    /// Observes notifications related to Freemium DBP (e.g., result polling complete or entry point activated),
+    /// and updates the promotion visibility state accordingly.
+    func observeFreemiumDBPNotifications() {
+        NotificationCenter.default.publisher(for: .freemiumDBPResultPollingComplete)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                Logger.freemiumDBP.debug("[Freemium DBP] Received Scan Results Notification")
+                self?.didDismissHomePagePromotion = false
+            }
+            .store(in: &cancellables)
+
+        NotificationCenter.default.publisher(for: .freemiumDBPEntryPointActivated)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                Logger.freemiumDBP.debug("[Freemium DBP] Received Entry Point Activation Notification")
+                self?.didDismissHomePagePromotion = true
+            }
+            .store(in: &cancellables)
```

Comment:
This type needs to react to two notification sources:

1. Scan results being sent by the `Polling` type
2. When a user opens Freemium DBP via the meatball menu item